controllers/chat/initChatFromPost.js

```diff
@@ -114,17 +114,26 @@ const initChatFromPost = async (req, res) => {
     try {
       if (!newChannel?.data?.name) {
         let channelType = CHANNEL_TYPE.CHAT;
+        let anonDetailInfo = {};
         if (userModel.is_anonymous || targetUserModel.is_anonymous) {
           channelType = CHANNEL_TYPE.ANONYMOUS;
+          anonDetailInfo = {
+            anon_user_info_color_code:
+              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code || null,
+            anon_user_info_color_name:
+              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name || null,
+            anon_user_info_emoji_code:
+              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code || null,
+            anon_user_info_emoji_name:
+              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name || null
+          };
         }
+
         await newChannel.updatePartial({
           set: {
             channel_type: channelType,
             name: [userModel?.username, targetUserModel?.username].join(', '),
-            anon_user_info_color_code: color.code,
-            anon_user_info_color_name: color.color,
-            anon_user_info_emoji_code: emoji.emoji,
-            anon_user_info_emoji_name: emoji.name,
+            ...anonDetailInfo,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a potential issue. If `newStateMemberWithAnonInfo[targetUserModel.user_id]` is `undefined`, it will cause an error when trying to access its properties. You should add a check for this before accessing the properties.

```diff
- anonDetailInfo = {
-   anon_user_info_color_code:
-     newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code || null,
-   anon_user_info_color_name:
-     newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name || null,
-   anon_user_info_emoji_code:
-     newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code || null,
-   anon_user_info_emoji_name:
-     newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name || null
- };
+ if (newStateMemberWithAnonInfo[targetUserModel.user_id]) {
+   anonDetailInfo = {
+     anon_user_info_color_code:
+       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code || null,
+     anon_user_info_color_name:
+       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name || null,
+     anon_user_info_emoji_code:
+       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code || null,
+     anon_user_info_emoji_name:
+       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name || null
+   };
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->